File: SourceCode/C/day054.c
/* Author: Ahmed A. M. ALOMAR
Date: February 23 - 2020
Description:
This program is an example of Scope */


int global_variable = 30;


int main()
{
  /* This function can access:
  1. The local_variable
  2. The Global_variable */
  int local_variable = 20;
  int adding = local_variable + global_variable;

  return 0;
}


void custom_function()
{
  /* This function can only access the another local variable */
  int another_local = 10;
}

File: SourceCode/Python/day054.py
# ------------------------------------ #
# Author: Ahmed A. M. ALOMAR
# Date: June 3 - 2020
#
# Description:
# * Example of Function Decorator
# ------------------------------------ #


# The decorator function is the heigher order function
def greetings(x):

  def inner():
    # extra functionality
    print("+++Nice to meet you! I am a decorator!")
    x()
    # extra functionality
    print("+++And this is another decorator after the function executes")

  return inner

def my_name():
  print("Ahmed A. M. Alomar")


greetings(my_name)()
